frontend/src/app/state/application/selectors/exercise.selectors.ts

```diff
@@ -212,34 +212,50 @@ export function createSelectByPredicate<E extends WithPosition>(
     );
 }
 
+export function createSelectBehaviorStates(simulatedRegionId: UUID) {
+    return createSelector(
+        createSelectSimulatedRegion(simulatedRegionId),
+        (simulatedRegion) =>
+            simulatedRegion.behaviors as ExerciseSimulationBehaviorState[]
```

Comment:
Is there a specific reason why you're casting to `ExerciseSimulationBehaviorState` instead of keeping the union type?

I think we're losing the ability to infer the type from the `type` property trough this change...

Reply:
Removed in 8f32e753ce819c5cd1c57db6bf76441dc62abd49